packages/core/src/domain/session/sessionCookieStore.ts

```diff
@@ -155,3 +162,28 @@ function isExpiredState(session: SessionState) {
 function clearSession(options: CookieOptions) {
   setCookie(SESSION_COOKIE_NAME, '', 0, options)
 }
+
+const MAX_AUDIT_ENTRIES = 50
+
+export class Audit {
+  constructor(private productKey: string, private entries: string[]) {}
+
+  addWrite(phase: string, session: SessionState) {
+    this.addEntry(phase, 'write', session)
+  }
+
+  addRead(phase: string, session: SessionState) {
+    this.addEntry(phase, 'read', session)
+  }
+
+  private addEntry(phase: string, operation: string, session: SessionState) {
+    if (this.entries.length < MAX_AUDIT_ENTRIES) {
+      const currentDate = new Date(timeStampNow())
+      this.entries.push(
+        `${currentDate.toLocaleTimeString()}.${currentDate.getMilliseconds()} - ${
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏format the string as ISO or at least non-locale, to avoid looking at translated or unexpected time formats

Reply:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏maybe the formatting can be done only if we find an inconsistency (optimization).

Reply:
- used `toTimeString`
- about optim, it'll only be done when FF is enabled so I'd say it is good enough